### PR TITLE
Specify resolver source to avoid using system resolver for CNAME adblocking (uplift to 1.20.x) 

### PR DESCRIPTION
**Pull request by bbondy**
Uplift of https://github.com/brave/brave-core/pull/7731 and Uplift of #7769

Resolves https://github.com/brave/brave-browser/issues/12575
Resolves brave/brave-browser#13527

Specify resolver source to avoid using system resolver for CNAME adblocking

Pre-approval checklist: 
- [x] You have tested your change on Nightly. 
- [ ] This contains text which needs to be translated. 
    - [ ] There are more than 7 days before the release. 
    - [ ] I've notified folks in #l10n on Slack that translations are needed. 
- [x] The PR milestones match the branch they are landing to. 

Pre-merge checklist: 
- [x] You have checked CI and the builds, lint, and tests all pass or are not related to your PR. 

Post-merge checklist: 
- [x] The associated issue milestone is set to the smallest version that the changes is landed on.

### CODE REVIEW

**Review by diracdeltas — APPROVED** [review]

**Review by kjozwiak — APPROVED** [review]
Uplift into `1.20.x` approved after deliberating with @brave/uplift-approvers.

**Pull request merged**
